url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/ChannelWriter.java

```diff
@@ -0,0 +1,86 @@
+package ccproxy;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.SocketChannel;
+import java.util.LinkedList;
+
+
+public class ChannelWriter {
+
+	Selector selector;
+	LinkedList<String> listToSend;
+	SocketChannel channel;
+	boolean writePaused=false;
+	ByteBuffer b;
+	
+	public ChannelWriter(SocketChannel channel, Selector selector) {
+		listToSend=new LinkedList<String>();
+		this.channel=channel;
+		b = ByteBuffer.allocateDirect(5000);
+		this.selector=selector;
+	}
+
+	
+	public void println(String s) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.